libs/javalib/src/mill/javalib/PublishModule.scala

```diff
@@ -600,6 +600,28 @@ trait PublishModule extends JavaModule { outer =>
       "Licenses" -> pom.licenses.map(l => s"${l.name} (${l.id})").mkString(",")
     )
   }
+
+  /**
+   * Creates a Maven repository directory with main and source artifacts for this module
+   *
+   * This results in a directory which contains things like
+   * `my/organization/my-module/VERSION/my-module-VERSION.{pom,jar}`.
+   * Such directories can later be used as Maven repositories during
+   * artifact resolution, or be merged together and then used as repository,
+   * for example.
+   */
+  def stagePublish: Task[PathRef] = Task {
```

Comment:
Let's put the `publish` prefix first, to namespace the task a bit better. `publishLocal` is already taken, maybe `publishStage`? `publishTestFolder`?